php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -1304,11 +1304,16 @@ public ListExpressionTree LIST_EXPRESSION() {
       );
   }
 
-  public ExpressionTree LIST_ELEMENT() {
-    return b.<ExpressionTree>nonterminal().is(
-      b.firstOf(
-        MEMBER_EXPRESSION(),
-        LIST_EXPRESSION()));
+  public ArrayAssignmentPatternElementTree LIST_ELEMENT() {
```

Comment:
@vilchik-elena It's slightly different:
```
[[$a], $b] = [[1], 2]; // ok
list(list($a), $b) = [[1], 2]; // ok
list([$a], $b) = [[1], 2]; // Fatal error: Cannot mix [] and list()
```
From the [RFC](https://wiki.php.net/rfc/short_list_syntax#details):
> Both due to implementation issues, and for consistency's sake, list() cannot be nested inside [], nor vice-versa

However, we could decide to support mixed list and [], that would be slightly easier for us.

Reply:
@pynicolas indeed, that's what I had in mind, we can be more permissive

Reply:
@vilchik-elena Changed in 9b154a59177915295c48ce7f07054e1b5a0f0a30